x/chainlink/types/types.go

```diff
@@ -37,3 +37,16 @@ func (dp DataProviders) Contains(addr sdk.Address) bool {
 
 	return false
 }
+
+func (dp DataProviders) Remove(addr sdk.Address) DataProviders {
+	m := make(map[string]*DataProvider)
+	for _, acc := range dp {
+		m[acc.GetAddress().String()] = acc
+	}
+	delete(m, addr.String())
+	s := make([]*DataProvider, 0, len(m))
+	for _, acc := range m {
+		s = append(s, acc)
+	}
+	return s
+}
```

Comment:
what if we just use one for loop like below?
```
s := make([]*DataProvider, 0, len(m)-1)
for _, acc := range dp {
   if !acc.GetAddress().Equal(addr){
        s = append(s, acc)
   }
}
return s
```

Reply:
You right, I have made this version to cover the case if the array contain the account more than one time, but I have added some checks after to avoid this.

Reply:
cool, ideally anteHandler should not allow duplicated data provider

Reply:
This is checked in ant handler here:

Add:
https://github.com/ChainSafe/chainlink-cosmos/blob/98b10f93ec1f218dcecca0850edcd21d08b9b91f/x/chainlink/ante/ante.go#L134-L136

Remove:
https://github.com/ChainSafe/chainlink-cosmos/blob/98b10f93ec1f218dcecca0850edcd21d08b9b91f/x/chainlink/ante/ante.go#L142-L144